crates/fluvio/src/producer/accumulator.rs

```diff
@@ -191,13 +253,13 @@ impl ProducerBatch {
     /// Add a record to the batch.
     /// Return ProducerError::BatchFull if record does not fit in the batch, so
     /// the RecordAccumulator can create more batches if needed.
-    fn push_record(&mut self, record: Record) -> Option<PartialFutureRecordMetadata> {
+    fn push_record(&mut self, record: Record) -> PushRecordStatus {
```

Comment:
I think it would be clearer to use a Result type instead of creating a PushRecordStatus enum: Result<PushRecordStatus, (BatchSize, Record)>

Reply:
Sure! I added one commit using result instead of a new enum